eng/versioning/external_dependencies.txt

```diff
@@ -115,8 +115,9 @@ com.microsoft.azure:azure-storage;8.0.0
 com.microsoft.azure:msal4j;1.3.0
 com.microsoft.azure:msal4j-persistence-extension;0.1
 com.sun.activation:jakarta.activation;1.2.1
-io.opentelemetry:opentelemetry-api;0.2.4
-io.opentelemetry:opentelemetry-sdk;0.2.4
+io.opentelemetry:opentelemetry-api;0.6.0
+io.opentelemetry:opentelemetry-sdk;0.6.0
+io.grpc:grpc-core:jar:1.23.0
```

Comment:
Is this required now? I know this will pull in a lot of other dependencies we don't commonly use such as GSON and OpenCensus.

cc: @JonathanGiles

Reply:
Yes, its been required for a while now. I was holding off on this change and there were ongoing conversations related to this here - https://github.com/open-telemetry/opentelemetry-java/issues/575. 
But not updating the version was making us fall behind on the latest updates from otel. 
So thought should finally have the updated versions for which this becomes required.

Would it be possible to include `grpc.core` without GSON and OpenCensus dependencies?

Reply:
Should be able to use the scoped down package which they use transitively in this, https://mvnrepository.com/artifact/io.grpc/grpc-context/1.28.0.